checkpoint/runner_test.go

```diff
@@ -371,7 +375,28 @@ func TestRunner_Generate_Error(t *testing.T) {
 		dir, err := afero.TempDir(fs, "", "Generate")
 		require.NoError(t, err)
 
-		err = checkpoint.Generate(context.Background(), fs, db, dir, snapshot, 2)
+		err = Generate(context.Background(), fs, db, dir, snapshot, 2)
 		require.Error(t, err)
 	})
 }
+
+func TestRunner_Generate_PreservesMarriageATX(t *testing.T) {
+	t.Parallel()
+	db := sql.InMemory()
+
+	require.NoError(t, accounts.Update(db, &types.Account{Address: types.Address{1, 1}}))
+
+	atx := &types.ActivationTx{
+		CommitmentATX: &types.ATXID{1, 2, 3, 4, 5},
+		MarriageATX:   &types.ATXID{6, 7, 8, 9},
+		SmesherID:     types.RandomNodeID(),
+		NumUnits:      4,
+	}
+	atx.SetID(types.RandomATXID())
+	require.NoError(t, atxs.Add(db, atx, types.AtxBlob{}))
+	require.NoError(t, atxs.SetUnits(db, atx.ID(), atx.SmesherID, atx.NumUnits))
+
+	checkpoint, err := checkpointDB(context.Background(), db, 0, 1)
```

Comment:
How difficult would it be to write this test only using exported functions and types rom the `checkpoint` package? In the long run we want to have all tests in `_tests` packages and shouldn't move in the other direction if we can avoid it.